common/src/main/java/dev/ftb/mods/ftbchunks/data/ClaimedChunk.java

```diff
@@ -98,34 +105,70 @@ public boolean allowExplosions() {
 	}
 
 	public void sendUpdateToAll() {
-		SendChunkPacket packet = new SendChunkPacket();
-		packet.dimension = pos.dimension;
-		packet.teamId = teamData.getTeamId();
-		packet.chunk = new SendChunkPacket.SingleChunk(System.currentTimeMillis(), pos.x, pos.z, this);
-		packet.sendToAll(teamData.manager.getMinecraftServer());
+		SendChunkPacket packet = new SendChunkPacket(pos.dimension, teamData.getTeamId(), new SendChunkPacket.SingleChunk(System.currentTimeMillis(), pos.x, pos.z, this));
+		ChunkSendingUtils.sendChunkToAll(teamData.manager.getMinecraftServer(), teamData, packet);
 	}
 
 	public void unload(CommandSourceStack source) {
 		if (isForceLoaded()) {
 			setForceLoadedTime(0L);
 			ClaimedChunkEvent.AFTER_UNLOAD.invoker().after(source, this);
 			teamData.save();
+			forceLoadExpiryTime = 0L;
 		}
 	}
 
 	public void unclaim(CommandSourceStack source, boolean sync) {
 		unload(source);
 
-		teamData.manager.claimedChunks.remove(pos);
+		teamData.manager.unregisterClaim(pos);
 		ClaimedChunkEvent.AFTER_UNCLAIM.invoker().after(source, this);
 		teamData.save();
 
 		if (sync) {
-			SendChunkPacket packet = new SendChunkPacket();
-			packet.dimension = pos.dimension;
-			packet.teamId = Util.NIL_UUID;
-			packet.chunk = new SendChunkPacket.SingleChunk(System.currentTimeMillis(), pos.x, pos.z, null);
+			SendChunkPacket packet = new SendChunkPacket(pos.dimension, Util.NIL_UUID, new SendChunkPacket.SingleChunk(System.currentTimeMillis(), pos.x, pos.z, null));
 			packet.sendToAll(source.getServer());
 		}
 	}
+
+	public long getForceLoadExpiryTime() {
+		return forceLoadExpiryTime;
+	}
+
+	public void setForceLoadExpiryTime(long forceLoadExpiryTime) {
+		this.forceLoadExpiryTime = forceLoadExpiryTime;
+		teamData.save();
+	}
+
+	public boolean hasExpired(long now) {
+		return forceLoadExpiryTime > 0L && forceLoadExpiryTime < now;
+	}
+
+	@Override
+	public String toString() {
+		return "[ " + pos.toString() + " - " + teamData + " ]";
+	}
+
+	public CompoundTag serializeNBT() {
+		SNBTCompoundTag o = new SNBTCompoundTag();
+		o.singleLine();
+		o.putInt("x", getPos().x);
```

Comment:
ChunkPos#toLong / fromLong?

Reply:
Probably, but yay for public fields, changing it will need a lot more changes throughout the mod. Definitely a good future cleanup though.